domain/tests/src/Unit/DomainValidatorTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\Tests\domain\Unit;
+
+use Drupal\Tests\UnitTestCase;
+use Drupal\domain\DomainValidator;
+
+/**
+ * Tests domain record validation.
+ *
+ * @group domain
+ */
+class DomainValidatorTest extends UnitTestCase {
+
+  /** @var \Drupal\domain\DomainValidator */
+  private $object;
+
+  /** @var \GuzzleHttp\Client */
+  public $client;
```

Comment:
Not required to be public. Just unsure of myself as I started working with dataProvider (which does need to be public)